sequencer/src/state.rs

```diff
@@ -402,83 +406,110 @@ impl<'a> StateCatchup for SqlStateCatchup<'a> {
 }
 
 pub async fn update_state_storage(
-    storage: Arc<RwLock<SqlDataSource<SeqTypes, Provider>>>,
+    storage: &mut SqlDataSource<SeqTypes, Provider>,
     instance: NodeState,
+    parent_leaf: LeafQueryData<SeqTypes>,
+    proposed_leaf: LeafQueryData<SeqTypes>,
```

Comment:
I think we don't need to take ownership of these things, `&` would do fine. This could save us from unnecessary cloning in `update_state_storage_loop`